pkg/kev/config/extensions.go

```diff
@@ -75,34 +78,149 @@ func DefaultK8SConfig() K8SConfiguration {
 			ReadinessProbe: DefaultReadinessProbe(),
 			Replicas:       1,
 		},
+		Service: Service{
+			Type: "None",
+		},
 	}
 }
 
 type k8sConfigOptions struct {
 	requireExtensions bool
+	disableValidation bool
 }
 
 // K8SCfgOption will modify parsing behaviour of the x-k8s extension.
 type K8SCfgOption func(*k8sConfigOptions)
 
+func DisableValidation() K8SCfgOption {
+	return func(kco *k8sConfigOptions) {
+		kco.disableValidation = true
+	}
+}
+
 // RequireExtensions will ensure that x-k8s is present and that it is validated.
 func RequireExtensions() K8SCfgOption {
 	return func(kco *k8sConfigOptions) {
 		kco.requireExtensions = true
 	}
 }
 
-// K8SCfgFromMap handles the extraction of the k8s-specific extension values from the top level map.
-func K8SCfgFromMap(m map[string]interface{}, opts ...K8SCfgOption) (K8SConfiguration, error) {
+func K8SCfgFromCompose(svc *composego.ServiceConfig) (K8SConfiguration, error) {
+	var cfg K8SConfiguration
+
+	cfg.Enabled = true
+	cfg.Workload.Type = WorkloadTypeFromCompose(svc)
+	cfg.Workload.Replicas = WorkloadReplicaFromCompose(svc)
+	cfg.Workload.RestartPolicy = WorkloadRestartPolicyFromCompose(svc)
+	cfg.Service.Type = NoService
+	if svc.Ports != nil {
+		cfg.Service.Type = ClusterIPService
+	}
+
+	cfg.Workload.LivenessProbe = LivenessProbeFromHealthcheck(svc.HealthCheck)
```

Comment:
This is not from compose service but from healthcheck

Reply:
OK, but `healthcheck` is part of compose service definition.

Reply:
maybe the misconception is with the other name, it should probably be FromService or ComposeSvc rather than just "compose". I think establishing a pattern of FromCompose when referring to svc and then FromCompose that takes an healthcheck is confusing and wrong

Reply:
How about `LivenessProbeFromCompose(svc)` then fetching what you need in that function internally?

Reply:
changed